src/containers/org/components/DataSetPermissionsCard.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * @flow
+ */
+
+import React, { useMemo, useState } from 'react';
+
+import _capitalize from 'lodash/capitalize';
+import styled from 'styled-components';
+import { faChevronDown, faChevronUp } from '@fortawesome/pro-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { List, Map } from 'immutable';
+import { Types } from 'lattice';
+import {
+  Card,
+  Colors,
+  IconButton,
+  Typography,
+} from 'lattice-ui-kit';
+import { useSelector } from 'react-redux';
+import type {
+  Ace,
+  EntitySet,
+  EntityType,
+  PermissionType,
+  Principal,
+  UUID,
+} from 'lattice';
+
+import { SpaceBetweenCardSegment } from '../../../components';
+import { selectEntitySetEntityType, selectPermissions } from '../../../core/redux/utils';
+
+const { NEUTRAL, PURPLE } = Colors;
+const { PermissionTypes } = Types;
+
+const MIXED_PERMISSIONS_LABEL :'Mixed Permissions' = 'Mixed Permissions';
+const NO_PERMISSIONS_LABEL :'No Permissions' = 'No Permissions';
+const ORDERED_PERMISSIONS = [
+  PermissionTypes.OWNER,
+  PermissionTypes.READ,
+  PermissionTypes.WRITE,
+  PermissionTypes.DISCOVER,
+  PermissionTypes.LINK,
+  PermissionTypes.MATERIALIZE,
+];
+
+const DataSetCard = styled(Card)`
+  background-color: ${NEUTRAL.N50};
+  border: none;
+`;
+
+const PermissionTypeCard = styled(Card)`
+  background-color: ${({ isSelected }) => (isSelected ? PURPLE.P00 : NEUTRAL.N00)};
+  border: 1px solid ${({ isSelected }) => (isSelected ? PURPLE.P300 : NEUTRAL.N00)};
+  margin-left: 24px;
+`;
+
+const ActionsWrapper = styled.div`
+  align-items: center;
+  display: flex;
+
+  > button {
+    margin-left: 16px;
+  }
+`;
+
+const DataSetPermissionsCard = ({
+  dataSet,
+  principal,
+} :{
+  dataSet :EntitySet;
+  principal :Principal;
+}) => {
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [targetPermissionType, setTargetPermissionType] = useState();
+
```

Comment:
missing initial value argument to `useState`

Reply:
though I suppose the implicit value is `undefined`

Reply:
The initial value is meant to be `undefined`. We don't know the "target" until the user clicks on it.